app/main.py

```diff
@@ -1,25 +1,65 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: str, name: str) -> None:
```

Comment:
The `owner` parameter in the `__set_name__` method should be of type `type`, not `str`. This method is called with the class owning the descriptor, not a string.